repose-aggregator/components/services/open-tracing-service/open-tracing-service-api/src/main/resources/META-INF/schema/config/open-tracing.xsd

```diff
@@ -0,0 +1,349 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+  Repose
+  _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+  Copyright (C) 2010 - 2015 Rackspace US, Inc.
+  _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+  =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+  -->
+
+
+<xs:schema xmlns:html="http://www.w3.org/1999/xhtml"
+           xmlns:xs="http://www.w3.org/2001/XMLSchema"
+           xmlns:vc="http://www.w3.org/2007/XMLSchema-versioning"
+           xmlns:xerces="http://xerces.apache.org"
+           xmlns:saxon="http://saxon.sf.net/"
+           xmlns="http://docs.openrepose.org/repose/open-tracing-service/v1.0"
+           elementFormDefault="qualified"
+           targetNamespace="http://docs.openrepose.org/repose/open-tracing-service/v1.0">
+
+    <xs:element name="open-tracing" type="OpenTracingConfig"/>
+
+    <xs:complexType name="OpenTracingConfig">
+        <xs:annotation>
+            <xs:documentation>
+                <html:p>OpenTracing configuration settings.</html:p>
+                <html:p>
+                    Please refer to the
+                    <html:a href="https://github.com/opentracing/specification/blob/master/specification.md">
+                        OpenTracing specification
+                    </html:a>
+                    for more details.
+                </html:p>
+            </xs:documentation>
+        </xs:annotation>
+
+        <xs:choice>
+            <xs:element name="jaeger" type="JaegerTracerConfig"/>
+        </xs:choice>
+
+        <xs:attribute name="service-name" type="xs:string" use="required">
+            <xs:annotation>
+                <xs:documentation>
+                    <html:p>
+                        The name of the service that is reporting traces.
+                        This value will be used to identify the service and should therefore be unique
+                        (e.g., identity-repose).
+                    </html:p>
+                </xs:documentation>
+            </xs:annotation>
+        </xs:attribute>
+
+        <xs:attribute name="span-context-key" type="xs:string" default="TBD">
```

Comment:
This still needs to be dealt with. As @dimtruck pointed out, there is a RFC meant to standardize on the header names:
https://w3c.github.io/distributed-tracing/report-trace-context.html

Of course, we have to wait for that to have its status bumped up, and for Jaeger (or other tracers) to support it. Especially since the proposal (at this time) calls for multiple headers being added to the request.

For now, Dimitry suggested defaulting to whatever the Jaeger library does, which seems fine, but I'm not sure how we would do that. Using a default here means that we always have some value to set when configuring the `Tracer`. We could allow this attribute to be null, but then if we want to add a default in the future doing so will probably be backwards incompatible. Or we could plan to never default this attribute. If the user does not configure it, we let the tracing library do its thing, otherwise, we write a single header with the configured header name. Once again, that will probably not suffice once that RFC status changes.